Can't attach empty fan curve data to fan

**Describe the bug**
The daemon crashes with the error "Can't attach empty fan curve data to fan system".
Might or might not be related to https://github.com/markusressel/fan2go/issues/465.

Did also fail on 0.13.0, but with no error.

**Additional context**
Using a self-written not-yet-in-the-kernel hwmon driver for Synology NAS devices.
It does not have an internal fan curve, so if there is no userspace daemon the fan speed stays static even under heavy cpu load, so it would really benefit from having this daemon.
Also
- it does not have a rpm sensor.
- set and get pwm value may differ. The driver controls the fan with percent (00-99 over serial bus).
e.g. if setting pwm to 120, it results in 121 as the driver can't control it that precise.
- pwm_mode is read-only

**Config**
```yaml
dbPath: /var/lib/fan2go/fan2go.db

logLevel: "info"

checkInterval: 20

sanityCheck:
  pwmValueChangedByThirdParty:
    enabled: false
  fanModeChangedByThirdParty:
    enabled: false

fans:
  - id: system
    hwmon:
      platform: synology_microp-virtual-0
      index: 3
    pwmMap: identity
    setPwmToGetPwmMap: identity
    minPwm: 120
    startPwm: 140
    maxPwm: 255
    neverStop: true
    curve: system_curve

sensors:
  - id: cpu_temp
    hwmon:
      platform: k10temp-pci-00c3
      index: 1

  - id: nvme1_temp
    disk:
      device: /dev/disk/by-path/pci-0000:02:00.0-nvme-1

  - id: sata1_temp
    disk:
      device: /dev/disk/by-path/pci-0000:07:00.0-ata-1

  - id: sata2_temp
    disk:
      device: /dev/disk/by-path/pci-0000:07:00.0-ata-2

  - id: sata3_temp
    disk:
      device: /dev/disk/by-path/pci-0000:09:00.0-ata-1

  - id: sata4_temp
    disk:
      device: /dev/disk/by-path/pci-0000:09:00.0-ata-2

  - id: nic_phy_temp
    hwmon:
      platform: trusted-pci-0100
      index: 1
  - id: nic_mac_temp
    hwmon:
      platform: trusted-pci-0100
      index: 2

curves:
  - id: system_curve
    function:
      type: maximum
      curves:
        - cpu_curve
        - nvme1_curve
        - sata1_curve
        - sata2_curve
        - sata3_curve
        - sata4_curve
        - nic_phy_curve
        - nic_mac_curve

  - id: cpu_curve
    linear:
      sensor: cpu_temp
      steps:
        - 0: 50%
        - 78: 70%
        - 86: 100%

  - id: nvme1_curve
    linear:
      sensor: nvme1_temp
      steps:
        - 0: 50%
        - 48: 60%
        - 56: 70%
        - 64: 100%

  - id: sata1_curve
    linear:
      sensor: sata1_temp
      steps:
        - 0: 50%
        - 46: 60%
        - 52: 70%
        - 55: 80%
        - 58: 100%

  - id: sata2_curve
    linear:
      sensor: sata2_temp
      steps:
        - 0: 50%
        - 46: 60%
        - 52: 70%
        - 55: 80%
        - 58: 100%

  - id: sata3_curve
    linear:
      sensor: sata3_temp
      steps:
        - 0: 50%
        - 46: 60%
        - 52: 70%
        - 55: 80%
        - 58: 100%

  - id: sata4_curve
    linear:
      sensor: sata4_temp
      steps:
        - 0: 50%
        - 46: 60%
        - 52: 70%
        - 55: 80%
        - 58: 100%

  - id: nic_phy_curve
    linear:
      sensor: nic_phy_temp
      steps:
        - 0: 50%
        - 72: 70%
        - 82: 100%

  - id: nic_mac_curve
    linear:
      sensor: nic_mac_temp
      steps:
        - 0: 50%
        - 72: 70%
        - 82: 100%
```

**Sensors**
```log
$ ./fan2go-linux-amd64-no-nvml detect
=========== hwmon: ============

> Platform: drivetemp-scsi-5-0
  Sensors  Channel  Label                       Value
           1        hwmon6/temp1 (temp1_input)  31000

> Platform: drivetemp-scsi-0-0
  Sensors  Channel  Label                       Value
           1        hwmon4/temp1 (temp1_input)  31000

> Platform: k10temp-pci-00c3
  Sensors  Channel  Label               Value
           1        Tctl (temp1_input)  42250

> Platform: synology_microp-virtual-0
  Fans     Index  PWM Channel  RPM Channel  Label         RPM  PWM  Mode
           1      1            N/A          System Fan 1  N/A  139  N/A
           2      2            N/A          System Fan 2  N/A  N/A  N/A
           3      1            N/A          hwmon0/pwm1   N/A  139  N/A
> Platform: drivetemp-scsi-6-0
  Sensors  Channel  Label                       Value
           1        hwmon7/temp1 (temp1_input)  31000

> Platform: drivetemp-scsi-1-0
  Sensors  Channel  Label                       Value
           1        hwmon5/temp1 (temp1_input)  31000

> Platform: trusted-pci-0100
  Sensors  Channel  Label                          Value
           1        PHY Temperature (temp1_input)  44000
           2        MAC Temperature (temp2_input)  44000

> Platform: nvme-pci-0200
  Sensors  Channel  Label                    Value
           1        Composite (temp1_input)  28850
           2        Sensor 1 (temp2_input)   28850
```

**Logs**
```log
$ sudo ./fan2go-linux-amd64-no-nvml -v
 INFO  Verbose output enabled
███████  █████  ███    ██ ██████   ██████   ██████
██      ██   ██ ████   ██      ██ ██       ██    ██
█████   ███████ ██ ██  ██  █████  ██   ███ ██    ██
██      ██   ██ ██  ██ ██ ██      ██    ██ ██    ██
██      ██   ██ ██   ████ ███████  ██████   ██████

 INFO  Version: 0.14.0
 INFO  Using configuration file at: /etc/fan2go/fan2go.yaml
  DEBUG   Feature 'temp1' is not a fan or pwm feature, ignoring.
  DEBUG   Feature 'temp1' is not a fan or pwm feature, ignoring.
  DEBUG   Feature 'temp1' is not a fan or pwm feature, ignoring.
  DEBUG   Feature 'curr1' is not a fan or pwm feature, ignoring.
  DEBUG   Feature 'temp1' is not a fan or pwm feature, ignoring.
  DEBUG   Feature 'temp1' is not a fan or pwm feature, ignoring.
  DEBUG   Feature 'temp1' is not a fan or pwm feature, ignoring.
  DEBUG   Feature 'temp2' is not a fan or pwm feature, ignoring.
  DEBUG   Feature 'temp1' is not a fan or pwm feature, ignoring.
  DEBUG   Feature 'temp2' is not a fan or pwm feature, ignoring.
 WARNING  WARN: cannot guarantee neverStop option on fan system, since it has no RPM input.
 INFO  Gathering sensor data for system...
 INFO  Fan system: Loading fan curve data...
 WARNING  Fan 'system' has not yet been analyzed, starting initialization sequence...
 INFO  Fan system: Using identity set→get PWM map
 INFO  Fan system: Using identity pwm map
  DEBUG   Target values with distinct PWM value of fan system: [0 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40 41 42 43 44 45 46 47 48 49 50 51 52 53 54 55 56 57 58 59 60 61 62 63 64 65 66 67 68 69 70 71 72 73 74 75 76 77 78 79 80 81 82 83 84 85 86 87 88 89 90 91 92 93 94 95 96 97 98 99 100 101 102 103 104 105 106 107 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251 252 253 254 255]
 INFO  Fan 'system' doesn't support RPM sensor, skipping fan curve measurement
 INFO  Fan system: Analysis boundaries detected (before config overrides): Start 255, Max 255
  ERROR   Can't attach empty fan curve data to fan system
  ERROR   Fan system: Failed to attach fan curve data: invalid argument
 INFO  Fan controller for fan system stopped.
 WARNING  Cannot send notification, missing env variable 'DISPLAY'!
panic: invalid argument

goroutine 181 [running]:
github.com/markusressel/fan2go/internal.RunDaemon.func7()
	/home/runner/work/fan2go/fan2go/internal/backend.go:144 +0x154
github.com/oklog/run.(*Group).Run.func1({0xc00038e240?, 0xc0003840d8?})
	/home/runner/go/pkg/mod/github.com/oklog/run@v1.2.0/group.go:38 +0x29
created by github.com/oklog/run.(*Group).Run in goroutine 1
	/home/runner/go/pkg/mod/github.com/oklog/run@v1.2.0/group.go:37 +0x67
```

**Server:**
- Distro: Debian Linux (headless)
- `uname -a`: 7.1+unreleased-amd64
- `sensors -v`: sensors version 3.6.2 with libsensors version 3.6.2
- `fan2go version`: 0.14.0

--- a/internal/fans/hwmon.go
+++ b/internal/fans/hwmon.go
@@ -123,11 +123,10 @@ func (fan *HwMonFan) GetFanRpmCurveData() *map[int]float64 {
 // AttachFanCurveData attaches fan curve data from persistence to a fan
 // Note: When the given data is incomplete, all values up until the highest
 // value in the given dataset will be interpolated linearly
-// returns os.ErrInvalid if curveData is void of any data
 func (fan *HwMonFan) AttachFanRpmCurveData(curveData *map[int]float64) (err error) {
 	if curveData == nil || len(*curveData) <= 0 {
-		ui.Error("Can't attach empty fan curve data to fan %s", fan.GetId())
-		return os.ErrInvalid
+		fan.FanCurveData = nil
+		return err
 	}
 
 	fan.FanCurveData = curveData

